### PR TITLE
a11y: handle left/right scrolls correctly

### PR DESCRIPTION
**Pull request by goderbauer**
... with awesome code-change to test ration :)

### CODE REVIEW

**Comment by goderbauer** [comment]
Looks like the test failure is exposing a real bug in our semantics algorithm, which I'll have to investigate before landing this.

**Comment by Hixie** [comment]
Should we close this PR in the meantime?

**Comment by goderbauer** [comment]
Fix for the test failure is in https://github.com/flutter/flutter/pull/11544.

File: packages/flutter/lib/src/widgets/gesture_detector.dart [hunk 1]
```diff
@@ -727,7 +727,7 @@ class _GestureSemantics extends SingleChildRenderObjectWidget {
     renderObject
       ..onTap = recognizers.containsKey(TapGestureRecognizer) ? _handleTap : null
       ..onLongPress = recognizers.containsKey(LongPressGestureRecognizer) ? _handleLongPress : null
-      ..onHorizontalDragUpdate = recognizers.containsKey(VerticalDragGestureRecognizer) ||
+      ..onHorizontalDragUpdate = recognizers.containsKey(HorizontalDragGestureRecognizer) ||
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/gesture_detector.dart]: !

**Comment by Hixie** [comment]
LGTM

**Pull request merged**